COLLABORATOR_GUIDE.md

```diff
@@ -0,0 +1,201 @@
+# io.js Collaborator Guide
+
```

Comment:
It might be cool to include a [ToC](http://npm.im/doctoc) for quick reference navigation.